Engine/DataFeeds/Enumerators/DividendEventProvider.cs

```diff
@@ -64,9 +65,16 @@ public IEnumerable<BaseData> GetEvents(NewTradableDateEventArgs eventArgs)
             {
                 if (_priceFactorRatio != null)
                 {
-                    var close = AuxiliaryDataEnumerator.GetRawClose(
-                        eventArgs.LastBaseData?.Price ?? 0,
-                        _config);
+                    decimal close;
+                    if (_referencePrice != null && _referencePrice != 0)
+                    {
+                        close = _referencePrice.Value;
+                    }
+                    else
+                    {
+                        close = eventArgs.LastRawPrice ?? 0;
+                    }
+
```

Comment:
Ideally, factor files can fill forward if no ref price and remove this branch and LastRawPrice.

The split warning can't get the factor file price as it'll introduce look-ahead bias. So sticking to as is:
https://github.com/QuantConnect/Lean/pull/4828/files#diff-a41c0f7b0b53c1edfae1df63b198fbb7R98

Reply:
Split and Dividend event provider will throw an exception when there is no reference price available 👍